### PR TITLE
[MRG+1] Allow passing Path objects to joblib.{dump,load}.

### PR DESCRIPTION
**Pull request by anntzer**
See #315.

### CODE REVIEW

File: joblib/numpy_pickle.py [hunk 1]
```diff
@@ -389,6 +393,8 @@ def dump(value, filename, compress=0, cache_size=100, protocol=None):
         # By default, if compress is enabled, we want to be using 3 by
         # default
         compress = 3
+    if Path and isinstance(filename, Path):
```
Comment (GaelVaroquaux) [inline joblib/numpy_pickle.py]: Nitpick: could you write this: "if Path is not None and isinstance(filename, Path)"

File: joblib/numpy_pickle.py [hunk 2]
```diff
@@ -440,6 +446,8 @@ def load(filename, mmap_mode=None):
     object might not match the original pickled object. Note that if the
     file was saved with compression, the arrays cannot be memmaped.
     """
+    if Path and isinstance(filename, Path):
```
Comment (GaelVaroquaux) [inline joblib/numpy_pickle.py]: Same nitpick.

**Comment by GaelVaroquaux** [comment]
Beside the 2 nitpicks, that are just cosmetics to make the intent of the code more explicit, this is :+1: from me. Thank you very much!

**Comment by anntzer** [comment]
fixed

**Comment by GaelVaroquaux** [comment]
LGTM. + 1 for merge 

Can we have another review from the core team?

**Comment by aabadie** [comment]
> Can we have another review from the core team?
Just had a look, it seems all good to me as well. Maybe a mention to this new feature could be added in CHANGES.rst.

**Comment by GaelVaroquaux** [comment]
> Maybe a mention to this new feature could be added in CHANGES.rst.

+1 for CHANGES.rst. Thanks for spotting that!

**Comment by anntzer** [comment]
Should I create a "0.9.5" entry?

**Comment by GaelVaroquaux** [comment]
> Should I create a "0.9.5" entry?

Yes, please

**Comment by anntzer** [comment]
Done.

**Comment by lesteve** [comment]
LGTM, merging thanks !

**Comment by lesteve** [comment]
@anntzer FYI you can use "Fix #issue_number" in the PR description so that merging a PR closes automatically the associated issue.

**Pull request merged**